File: app/src/main/java/com/example/win/easy/dagger/module/ThreadModule.java
package com.example.win.easy.dagger.module;

import com.example.win.easy.thread.AppExecutors;

import java.util.concurrent.Executor;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ThreadModule {

    @Provides
    @Singleton
    @Named("dbAccess") static Executor provideDiskIO(AppExecutors appExecutors){
        return appExecutors.diskIO();
    }

    @Provides
    @Singleton
    @Named("mainThread") static Executor provideMainThread(AppExecutors appExecutors){
        return appExecutors.mainThread();
    }
}

File: app/src/main/java/com/example/win/easy/dagger/module/ThreadModule.kt
package com.example.win.easy.dagger.module

import dagger.Module
import dagger.Provides
import kotlinx.coroutines.ExecutorCoroutineDispatcher
import kotlinx.coroutines.ObsoleteCoroutinesApi
import kotlinx.coroutines.newSingleThreadContext
import java.util.concurrent.Executor
import java.util.concurrent.ExecutorService
import javax.inject.Named
import javax.inject.Singleton

@Module
object ThreadModule {

    @ObsoleteCoroutinesApi
    @JvmStatic
    @Provides
    @Singleton
    fun provideDownloadDispatcher() = newSingleThreadContext("DownloadThread")
}